src/main/java/org/broadinstitute/consent/http/service/MetricsService.java

```diff
@@ -223,9 +232,10 @@ public DatasetMetrics generateDatasetMetrics(Integer datasetId) {
     //find dars with the given datasetId in their list of datasetIds, datasetId is a String so it can be converted to jsonb in query
     //convert all dars into smaller objects that only contain the information needed
     List<DataAccessRequest> dars = darDAO.findAllDataAccessRequestsByDatasetId(Integer.toString(datasetId));
-    List<DarMetricsSummary> darInfo = dars.stream().map(dar ->
-      new DarMetricsSummary(dar))
-      .collect(Collectors.toList());
+    List<DarMetricsSummary> darInfo = dars.stream().map(dar -> {
+              DarCollection collection = darCollectionDAO.findDARCollectionByCollectionId(dar.getCollectionId());
```

Comment:
Same dao-loop feedback here. Consider this optional as well - it's a very under-used code path

Reply:
Here's another way to do what you're trying to do here with a single dao call as opposed to N dao calls:
```
    List<Integer> darCollectionIds = dars.stream().map(DataAccessRequest::getCollectionId).collect(Collectors.toList());
    List<DarCollection> darCollections = darCollectionDAO.findDARCollectionByCollectionIds(darCollectionIds);
    Map<Integer, DarCollection> collectionMap = darCollections.stream().collect(Collectors.toMap(DarCollection::getDarCollectionId, Function.identity()));
    List<DarMetricsSummary> darInfo = dars.stream().map(dar -> {
              DarCollection collection = collectionMap.get(dar.getCollectionId());
              return new DarMetricsSummary(dar, collection.getDarCode());
            }).collect(Collectors.toList());
```

Reply:
Thanks for this suggestion! It has been helpful!